src/admin/containers/AddonPage/index.js

```diff
@@ -132,7 +132,7 @@ function mapStateToProps(state, ownProps) {
 const CurrentAddonPage = asyncConnect([{
   key: 'AddonPage',
   deferred: true,
-  promise: loadAddonIfNeeded,
+  promise: safePromise(loadAddonIfNeeded),
```

Comment:
I wonder if we should have a helper function. `deferred: true, promise: safePromise(realPromiseMethod)` seems like a pattern we use a lot.